charts/newrelic-infrastructure-v3/templates/kubelet/daemonset.yaml

```diff
@@ -161,6 +161,17 @@ spec:
             {{- with .Values.kubelet.extraEnv }}
             {{- toYaml . | nindent 12 }}
             {{- end }}
+            {{- /* Needed to populate clustername in integration metrics */}}
+            - name: "CLUSTER_NAME"
+              value: {{ include "newrelic.cluster" . | required "cluster name must be set" }}
+            - name: "NRIA_PASSTHROUGH_ENVIRONMENT"
+              value: "CLUSTER_NAME"
```

Comment:
This could be dangerous as some integrations might have a `CLUSTER_NAME` argument that is not the K8s cluster name, but the integration cluster name: E.g. the Kafka cluster name, haproxy cluster name, etc.

Reply:
Are we sure that is not the intended behavior, and what we are doing right now a bug?

Reminds me of this: https://github.com/newrelic/nri-haproxy/issues/28